src/components/Form/FieldUpload/docs.stories.tsx

```diff
@@ -0,0 +1,115 @@
+import { Box, Button, Stack } from '@chakra-ui/react';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useForm } from 'react-hook-form';
+import { z } from 'zod';
+
+import { zFieldUploadValue } from '@/files/schemas';
+
+import { Form, FormField } from '../';
+import { FieldUploadPreview } from './FieldUploadPreview';
+import { useFieldUploadFileFromUrl } from './utils';
+
+export default {
+  title: 'Form/FieldUpload',
+};
+
+type FormSchema = z.infer<ReturnType<typeof zFormSchema>>;
+const zFormSchema = () =>
+  z.object({
+    file: zFieldUploadValue().optional(),
+  });
+
+const formOptions = {
+  mode: 'onBlur',
+  resolver: zodResolver(zFormSchema()),
+} as const;
+
+export const Default = () => {
+  const form = useForm<FormSchema>({
+    defaultValues: {
+      file: undefined,
+    },
+    ...formOptions,
+  });
+
+  return (
+    <Form {...form} onSubmit={(values) => console.log(values)}>
+      <Stack spacing={4}>
+        <FormField
+          control={form.control}
```

Comment:
optional, we can remove it :) this props is needed only if it isn't used inside a FormProvider (aka. our Form component)
cf. https://react-hook-form.com/docs/usecontroller/controller